File: springboot-boilerplate/src/main/java/com/grepp/spring/app/controller/websocket/team/payload/ChatMessage.java
package com.grepp.spring.app.controller.websocket.team.payload;

import java.time.OffsetDateTime;
import java.time.format.DateTimeFormatter;
import lombok.Data;

@Data
public class ChatMessage {

    private String sender;
    private String content;
    private final String createdAt = OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
    private final String time = OffsetDateTime.now().format(DateTimeFormatter.ofPattern("HH:mm"));

}

File: chat-service-B/src/main/kotlin/com/grepp/spring/app/controller/websocket/team/payload/ChatMessage.kt
package com.grepp.spring.app.controller.websocket.team.payload

import java.time.OffsetDateTime
import java.time.format.DateTimeFormatter

data class ChatMessage(
    var sender: String,
    val content: String,
    val createdAt: String =
        OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME),
    val time: String = OffsetDateTime.now().format(DateTimeFormatter.ofPattern("HH:mm"))
) {
}
